src/transport/SecureSessionMgr.cpp

```diff
@@ -228,15 +233,6 @@ CHIP_ERROR SecureSessionMgr::SendMessage(SecureSessionHandle session, PayloadHea
     ChipLogProgress(Inet, "Secure msg send status %s", ErrorStr(err));
     SuccessOrExit(err);
 
-    if (bufferRetainSlot != nullptr)
-    {
-        // Rewind the start and len of the buffer back to pre-send state for following possible retransmition.
-        encryptedMsg->SetStart(msgStart);
-        encryptedMsg->SetDataLength(msgLen);
-
-        (*bufferRetainSlot) = std::move(encryptedMsg);
-    }
-
 exit:
```

Comment:
Why we need to adjust this logic? the previous logic is we rewind the buffer back  to pre-send state after transmission.

Now, we do the rewind first and then transmit, it looks not right to me.